### DISCUSSION TITLE
Create dimmer button with "Held for 1000ms actions"

### DISCUSSION THREAD

**Discussion by Lulu2409** [bitfocus/companion#2749] [Q&A] [answered]
Hello everyone,
I have the following problem, which I would like to solve with Companion.
I have a lamp that is controlled via ARTNET.
I want to create two buttons.
once: dim on and up. and once: OFF and dim down.
These buttons should have the following behavior:
If I press the "on" button briefly, a set value (100%) should be set. If I hold it for a long time, the lamp should slowly dim up as long as I press it.
same thing with the “OFF” button:
If I press it briefly, the lamp should go out and if I press it long, the lamp should slowly dim down.
I wanted to solve the whole thing using the “Short release actions” and the “Held for 1000ms actions”. Unfortunately, this doesn't seem to work when dimming.
Maybe someone can help me with this and provide a solution.
Thank you!
![grafik](https://github.com/bitfocus/companion/assets/135103496/65147a54-4d22-4ea1-a414-c58f138dc567)

![grafik](https://github.com/bitfocus/companion/assets/135103496/251354f8-5564-4c5b-ab53-bf9b5b714d91)

**Comment by Julusian** [comment, marked as answer]
I think you are on the right track, but the `held for 1000ms actions` will only be executed the once, so after 1s it will be decreasing the value by 10 and then stop.
Instead, I would suggest using a `set value` (or `set complex value`) action in the `held for 1000ms actions` group. If you give a value for `fade time` then it will slowly change to the new value you specify.

It sounds like you want it to then stop fading at whatever point it got to when the button is released, which you could do by adding a second delayed group `release after 1001ms`, which does an `offset value` with a `value change` of 0. This will replace the currently running fade inside the module with a new 'fade' which does nothing, causing it to stop.

For clarity, a screenshot of what I think you should try (I havent tested this)
![image](https://github.com/bitfocus/companion/assets/1327476/dd08867b-2390-45db-a147-d2ce722d1db0)

**Reply by Lulu2409** [reply]
Thank you
That sounds logical.
I just tested it and it works very well.
The dimmer speed can now also be adjusted via the fade time.
Thank you for this great tip and the quick help!
Best regards
